### PR TITLE
test(indexing-service): migrate ingestion and parallel tests to JUnit 5

### PR DESCRIPTION
**Pull request by FrankChen021**
## Summary

- Migrate the ingestion-task and parallel-indexing tests in `indexing-service` to JUnit 5.
- Replace JUnit 4 assertions, lifecycle annotations, rules, Hamcrest matchers, and temporary-folder APIs in this batch with Jupiter/AssertJ and Druid's existing temporary-directory utilities.
- Add the Jupiter parameterized-test and Mockito Jupiter test dependencies required by the migrated tests.
- Keep the remaining JUnit 4 lifecycle annotations in the shared `IngestionTestBase` only as a compatibility bridge for legacy tests outside this batch.
- Keep this batch independent from the Task Actions/Compaction, common-support, Overlord, seekable-stream, and worker batches.

## Validation

- Two fresh focused reruns of `ParallelIndexSupervisorTaskResourceTest`: 1 test each, 0 failures/errors.
- Test compilation passed.
- Static module verification passed with Checkstyle, PMD, Enforcer, forbidden APIs, packaging, and install.
- Legacy/prohibited usage scans and `git diff --check` passed.

## Tracking

Part of [#13948](https://github.com/apache/druid/issues/13948).

Related migration PRs: [#19910](https://github.com/apache/druid/pull/19910), [#19920](https://github.com/apache/druid/pull/19920), [#19922](https://github.com/apache/druid/pull/19922), [#19923](https://github.com/apache/druid/pull/19923), and [#19925](https://github.com/apache/druid/pull/19925).

### CODE REVIEW

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/IngestionTestBase.java [hunk 1]
```diff
@@ -468,7 +475,8 @@ public ListenableFuture<TaskStatus> run(Task task)
         lockbox.add(task);
         taskStorage.insert(task, TaskStatus.running(task.getId()));
         taskActionClient = createActionClient(task);
-        taskReportsFile = temporaryFolder.newFile(
+        taskReportsFile = new File(
+            temporaryFolder,
             StringUtils.format("ingestionTestBase-%s.json", System.currentTimeMillis())
         );
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply indexing-service/src/test/java/org/apache/druid/indexing/common/task/IngestionTestBase.java:484]: Evaluated and intentionally left the production path unchanged. The parallel test path writes its report directly to reportsFile in AbstractParallelIndexSupervisorTaskTest; taskReportsFile belongs to the separate legacy TestTaskRunner helper. The focused CompactionTaskParallelRunTest and related tests pass.

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/ShardSpecsTest.java [hunk 2]
```diff
@@ -84,7 +84,7 @@ public void testShardSpecSelectionWithNullPartitionDimension()
     ShardSpec spec4 = shardSpecs.getShardSpec(Intervals.of("2014-01-01T00:00:00.000Z/2014-01-02T00:00:00.000Z"), row2);
     ShardSpec spec5 = shardSpecs.getShardSpec(Intervals.of("2014-01-01T00:00:00.000Z/2014-01-02T00:00:00.000Z"), row3);
 
-    Assert.assertSame(true, spec3 == spec4);
-    Assert.assertSame(false, spec3 == spec5);
+    Assertions.assertSame(true, spec3 == spec4);
+    Assertions.assertSame(false, spec3 == spec5);
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply indexing-service/src/test/java/org/apache/druid/indexing/common/task/ShardSpecsTest.java]: Fixed in commit a5780d1c93: replaced assertSame(true, ...) and assertSame(false, ...) with assertSame(spec3, spec4) and assertNotSame(spec3, spec5), so the test directly checks ShardSpec identity.

File: indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/AbstractParallelIndexSupervisorTaskTest.java [hunk 3]
```diff
@@ -167,11 +165,27 @@
 
   private static final Logger LOG = new Logger(AbstractParallelIndexSupervisorTaskTest.class);
 
-  @Rule
-  public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+  protected final File temporaryFolder = FileUtils.createTempDir();
```
[the remark this answers is not in the document]
Comment (FrankChen021) [reply indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/AbstractParallelIndexSupervisorTaskTest.java]: Fixed in commit a5780d1c93: renamed the subclass field to parallelTemporaryFolder and updated its uses, preserving its separate cleanup lifecycle while removing the field shadowing.

**Review by FrankChen021 — COMMENTED** [review]
reviewed all changes made by codex, LGTM

**Review by FrankChen021 — COMMENTED** [review]
I have reviewed the code for correctness, edge cases, concurrency, and integration risks; no issues found.

Reviewed 63 of 63 changed files.

Validation: git diff --check and the static migration scan passed; tests were not run.



---

This is an automated review by Codex GPT-5.6-Luna(max)

**Review by GWphua — APPROVED** [review]
